packages/core/src/config.ts

```diff
@@ -1,3 +1,75 @@
+export const ExternalSecretSystems = {
+  aws_ssm_parameter_store: 'aws_ssm_parameter_store',
+} as const;
+
+export type ExternalSecretSystem = keyof typeof ExternalSecretSystems;
+export type ExternalSecretPrimitive = string | boolean | number;
+export type ExternalSecretPrimitiveType = 'string' | 'boolean' | 'number';
+
+export type ExternalSecret<T extends ExternalSecretPrimitiveType = ExternalSecretPrimitiveType> = {
+  system: ExternalSecretSystem;
+  key: string;
+  type: T;
+};
+
+export interface MedplumSourceInfraConfig {
+  name: string | ExternalSecret<'string'>;
```

Comment:
Can we create a type for this common union?

```ts
type ValueOrExternalSecret<T> = T | ExternalSecret<T>;
```

Reply:
Good idea